app/src/main/java/com/poupa/vinylmusicplayer/provider/StoreLoader.java

```diff
@@ -36,11 +36,14 @@
         ArrayList<Long> orphanSongIds = new ArrayList<>();
 
         ArrayList<Song> songs = new ArrayList<>();
-        final boolean isStale = discography.isStale(); // Must capture this before the discog.getSong
+
+        // Spinning wait for the media store refresh to finish
+        while (discography.isStale()) {}
```

Comment:
I'm wondering if this spinning wait is creating any visible penalty to the start time of the app?

Reply:
Didn't notice it on my phone but it is a recent one, you could peraphs test this with an old phone or with the phone emulator of android studio to exaggerated the boot time